tensorflow/core/kernels/data/experimental/snapshot_dataset_op.cc

```diff
@@ -547,8 +551,6 @@ SnapshotDatasetV2Op::Dataset::Iterator::Reader::Reader(const Params& params,
                                                        int64 start_index)
     : DatasetIterator<Dataset>(params), start_index_(start_index) {}
 
-SnapshotDatasetV2Op::Dataset::Iterator::Reader::~Reader() { input_->Unref(); }
```

Comment:
Thanks for the PR. I think you also need to remove the declaration on Line 204 to make it work.

Reply:
You're right. Thank you